src/components/TextInput.tsx

```diff
@@ -14,46 +13,36 @@ type TextInputProps = {
   value: string;
   type: string;
   isError: boolean;
-  backgroundColor?: string;
   className?: string;
   onChange: (value: string) => void;
 };
 
 export const TextInput = (props: TextInputProps) => {
-  const [active, setActive] = useState(false);
-
   return (
-    <div className={cn("relative h-11", props.className)}>
+    <div className={cn("relative h-11 ", props.className)}>
       <h2
         className={twMerge(
           cn(
-            "absolute left-3 transition-all pointer-events-none z-10",
-            props.backgroundColor,
+            "absolute left-3 transition-all pointer-events-none z-10 bg-background px-2",
           ),
-          active || props.value != ""
-            ? "top-[-0.75rem] px-2 text-sm  text-blue-500"
-            : "top-1.5 cursor-text text-gray-500",
-          props.isError &&
-            "absolute left-3 top-[-0.75rem] px-2 text-sm  text-red-500",
+          props.value != ""
+            ? "top-[-0.75rem] text-sm text-blue-500"
+            : "top-1.5 cursor-text text-tertiary/70 dark:text-white/70",
+          props.isError && "left-3 top-[-0.75rem] px-2 text-sm text-red-500",
         )}
       >
         {props.label}
       </h2>
-
       <input
         type={props.type}
         defaultValue={props.value}
         onChange={(e) => props.onChange(e.target.value)}
-        onFocus={() => setActive(true)}
-        onBlur={() => setActive(false)}
+        value={props.value}
         className={twMerge(
           cn(
-            "w-full rounded p-2 border focus:border-2 outline-none h-full",
-            props.backgroundColor,
+            "w-full rounded p-2 border focus:border-2 outline-none h-full bg-background text-tertiary dark:text-white/70 InputAdjacentText",
+            "border-blue-400 focus:border-blue-500",
           ),
-          active
-            ? "border-blue-500 focus:border-blue-500"
-            : "border-blue-400 focus:border-blue-400",
           props.isError && "border-red-500 focus:border-red-500",
         )}
         required
```

Comment:
if you extends the TextInputProps interface correctly, you should be able to just destructure the remaining props like so.
```suggestion
      <input
        className={twMerge(
          cn(
            "w-full rounded p-2 border focus:border-2 outline-none h-full bg-background text-tertiary dark:text-white/70 InputAdjacentText",
            "border-blue-400 focus:border-blue-500",
          ),
          props.isError && "border-red-500 focus:border-red-500",
        )}
        {...props}
```